pkg/target/host.go

```diff
@@ -235,12 +234,16 @@ func (h *Host) getConnectionToHost() (net.Conn, error) {
 	return conn, err
 }
 
-func (h *Host) checkUpdateHostStatus(hostStatus chan HostStatus) {
+func (h *Host) checkUpdateHostStatus() {
 	conn, _ := h.getConnectionToHost()
 	if conn != nil {
-		hostStatus <- HostStatus{Host: h, Status: true, sigCh: make(chan struct{})}
+		h.Available.Store(true)
 		_ = conn.Close()
 	} else {
-		hostStatus <- HostStatus{Host: h, Status: false, sigCh: make(chan struct{})}
+		if h.Available.Load() {
```

Comment:
```suggestion
		if h.Available.CAS(true, false) {
```

And then we don't need
```
 		h.Available.Store(false) 
```